File: C/Data Structure/Sorting/program3.c
//Insertion Sort

#include<stdio.h>
#include<stdlib.h>

void Display(int Arr[], int iLength)
{
    int iCnt = 0;
    
    printf("Elements of array are : \n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        printf("%d\t",Arr[iCnt]);
    }
    printf("\n");
}

void InsertionSort(int Arr[], int iLength)
{
    int i = 0, j= 0, selected = 0;

    for(i = 1; i < iLength; i++)
    {
        for(j = i-1, selected = Arr[i]; (j >=0) && (Arr[j] > selected) ;j--)
        {
            Arr[j+1] = Arr[j];
        }
        Arr[j+1] = selected;
    }
}

int main()
{
    int *ptr = NULL;
    int iCnt = 0, iSize = 0;

    printf("Enter the number of elements\n");
    scanf("%d",&iSize);

    ptr = (int *)malloc(iSize * sizeof(int));

   printf("Enter elements : \n");
   for(iCnt = 0; iCnt < iSize; iCnt++ )
   {
        scanf("%d",&ptr[iCnt]);
   }

   Display(ptr, iSize);

    InsertionSort(ptr, iSize);
    printf("Data after sorting\n");
    Display(ptr, iSize);

    free(ptr);

    return 0;
}

File: Java/Data Structure/Sorting/program3.java
//Insertion Sort

import java.util.*;

class ArrayX
{
    public int Arr[];

    public ArrayX(int iSize)
    {
        Arr = new int[iSize];
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter elements");
        for(int i= 0;i <Arr.length; i++)
        {
            Arr[i] = sobj.nextInt();
        }
    }

    public void Display()
   {
        System.out.println("Elements of the array are");
        for(int i= 0;i <Arr.length; i++)
        {
           System.out.print(Arr[i]+"  ");
        }
        System.out.println();
   }

   public void InsertionSort()
   {
        int i = 0, j= 0, selected = 0;

        for(i = 1; i < Arr.length; i++)
        {
            for(j = i-1, selected = Arr[i]; (j >=0) && (Arr[j] > selected) ;j--)
            {
                Arr[j+1] = Arr[j];
            }
            Arr[j+1] = selected;
        }
   }

}

class program3
{
    public static void main(String arr[])
    {
            Scanner sobj = new Scanner(System.in);

            System.out.println("Enter the size of array : ");
            int size = sobj.nextInt();

            ArrayX aobj = new ArrayX(size);
            aobj.Accept();
            aobj.Display();

            aobj.InsertionSort();
            System.out.println("Data after sorting");
            aobj.Display();
    }
}